### PR TITLE
fcitx5: 5.0.4 -> 5.0.5

### PR DESCRIPTION
**Pull request by poscat0x04**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 115386` at e185a0e9 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/390283e8af72a421f272e094d21344f9">libsForQt512.fcitx5-qt</a></li></li>
  </ul>
</details>
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/a87c4263ef1eda4e7e8c059e6f4d9c1d">fcitx5</a></li></li>
    <li><a href="https://gist.github.com/ec4772e063512e23460c20670baaf46c">fcitx5-chinese-addons</a></li></li>
    <li><a href="https://gist.github.com/b427aa9d2a87f730cd22bd5ca48098a1">fcitx5-configtool</a></li></li>
    <li><a href="https://gist.github.com/05c16ad90ba38b3b86eb9b034a3dd9f7">fcitx5-gtk</a></li></li>
    <li><a href="https://gist.github.com/931ba381a433fd85848a24b3e8c1a986">fcitx5-lua</a></li></li>
    <li><a href="https://gist.github.com/17a83a0959dcb874eb0c39f1154f55cb">fcitx5-mozc</a></li></li>
    <li><a href="https://gist.github.com/c57a43f425231231ff64b4e8c070243d">fcitx5-rime</a></li></li>
    <li><a href="https://gist.github.com/4468781dc0bfdc2e9a48aca080dc122b">fcitx5-table-extra</a></li></li>
    <li><a href="https://gist.github.com/03858e1e8311ce830da8bc7ca6032de8">fcitx5-table-other</a></li></li>
    <li>fcitx5-with-addons</li>
    <li><a href="https://gist.github.com/e54629ac6247833dfed61b644614c3fb">libime</a></li></li>
    <li><a href="https://gist.github.com/e80a1804107e35876d6fa7154492dc05">libsForQt5.fcitx5-qt</a></li> (libsForQt515.fcitx5-qt ,plasma5Packages.fcitx5-qt)</li>
    <li><a href="https://gist.github.com/d2faf8835a6c0334b91ffb0951ce8456">libsForQt514.fcitx5-qt</a></li></li>
  </ul>
</details>

File: pkgs/tools/inputmethods/fcitx5/fcitx5-gtk.nix [hunk 1]
```diff
@@ -19,23 +20,25 @@
...
     repo = "fcitx5-gtk";
     rev = version;
-    sha256 = "sha256-+BzXbZyzC3fvLqysufblk0zK9fAg5jslVdm/v3jz4B4=";
+    sha256 = "sha256-rP0L7D7VdbNSmwQhnS5tvQT4FS3qpOnuBTQQp82O5fE=";
   };
 
   cmakeFlags = [
     "-DGOBJECT_INTROSPECTION_GIRDIR=share/gir-1.0"
     "-DGOBJECT_INTROSPECTION_TYPELIBDIR=lib/girepository-1.0"
+    "-DCMAKE_CXX_STANDARD_INCLUDE_DIRECTORIES=${glib.dev}/include/gio-unix-2.0"
```
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/fcitx5/fcitx5-gtk.nix:41]:
```suggestion
    "-DCMAKE_CXX_STANDARD_INCLUDE_DIRECTORIES=${lib.getDev glib}/include/gio-unix-2.0"
```

File: pkgs/tools/inputmethods/fcitx5/update.py [hunk 2]
```diff
@@ -12,8 +12,7 @@
 OWNER = "fcitx"
 
 def get_latest_tag(repo, owner=OWNER):
-    r = requests.get( 'https://api.github.com/repos/{}/{}/tags'.format(owner,repo)
-                    , auth=('poscat', 'db5e6fd16d0eb8c36385d3d944e058a1178b4265'))
```
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/fcitx5/update.py:16]: Did you revoke that token?

**Comment by SuperSandro2000** [comment]
I think we should mark it broken for qt512 if there is no upstream patch.

File: pkgs/tools/inputmethods/fcitx5/update.py (hunk 2 above)
Comment (poscat0x04) [reply pkgs/tools/inputmethods/fcitx5/update.py:16]: It seems that GitHub scans commits for API tokens and revokes those found :facepalm:. The reason I added the token initially is that I was hitting the rate limit (due to debugging). Now that I no longer need to debug the script, I shouldn't be hitting the rate limit anymore.

**Comment by poscat0x04** [comment]
hmm, how do I mark it only for qt 5.12?

**Comment by SuperSandro2000** [comment]
> hmm, how do I mark it only for qt 5.12?

See https://github.com/NixOS/nixpkgs/blob/bcf1cb07125b84c3f49f62b0b4e55626e85eb2e7/pkgs/applications/kde/kdenetwork-filesharing.nix#L12

**Comment by poscat0x04** [comment]
done

**Comment by poscat0x04** [comment]
ping @SuperSandro2000

File: pkgs/tools/inputmethods/fcitx5/fcitx5-qt.nix [hunk 3]
```diff
@@ -48,6 +48,7 @@ mkDerivation rec {
     homepage = "https://github.com/fcitx/fcitx5-qt";
     license = with licenses; [ lgpl21Plus bsd3 ];
     maintainers = with maintainers; [ poscat ];
+    broken = lib.versionOlder qtbase.version "5.13";
```
Comment (oxalica) [inline pkgs/tools/inputmethods/fcitx5/fcitx5-qt.nix:51]: It is fixable for qt 5.12. Check these patches.
https://github.com/NixOS/nixpkgs/pull/116748/files#diff-ea7a3a933b8f107b11a9c18c604e990761e6c6f590170d5420eb28838c30adc1R25

**Review by oxalica — CHANGES_REQUESTED** [review]

**Comment by poscat0x04** [comment]
@oxalica I don't have time to test that patch rn. Do you mind me adding you as a maintainer for fcitx5 (so you can apply that patch afterward)?

**Comment by oxalica** [comment]
@poscat0x04 Fine. Since the default `libsForQt5` refers to Qt 5.15.2 now, it should work well with plasma and the compatibility should be less important.

**Pull request closed without merging**